src/plugins/plugin-connect.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var urlParse = require('url').parse;
+
+var SUPPORTED_VERSIONS = '3.x';
+
+function middleware(api, req, res, next) {
+  var originalEnd = res.end;
```

Comment:
I don't think there is a reason for it to be hoisted.  I've moved it down to the redefinition of `res.end`.